meta-client/src/main/java/org/apache/helix/metaclient/recipes/lock/LockClientInterface.java

```diff
@@ -0,0 +1,58 @@
+package org.apache.helix.metaclient.recipes.lock;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import org.apache.helix.metaclient.api.MetaClientInterface;
+
+public interface LockClientInterface {
+  /**
+   * Acquires a lock at key.
+   * @param key key to identify the entry
+   * @param info Metadata of the lock
+   * @param mode EntryMode identifying if the entry will be deleted upon client disconnect
```

Comment:
Maybe we can mention what are the entry modes, might be easier to understand.

Reply:
Sounds good, addressed!